p2p/dialer.go

```diff
@@ -110,6 +114,15 @@ func (m *dialer) exchangeConfig(consumerID, providerID identity.Identity, servic
 	}
 	defer brokerConn.Close()
 
+	ready := make(chan bool)
+	_, err = brokerConn.Subscribe(channelHandlersReadySubject(providerID, serviceType), func(msg *nats_lib.Msg) {
+		defer close(ready)
```

Comment:
I have just realized that multiple calls for p2p endpoint will try to close channel multiple times, which will panic. 
We need to add something like `sync.Once:
```suggestion
	ready := make(chan bool)
	var once sync.Once
	_, err = brokerConn.Subscribe(channelHandlersReadySubject(providerID, serviceType), func(msg *nats_lib.Msg) {
		defer once.Do(func() { close(ready) })
```